File: PythonExample/example01/cwa_api_client.py


class CWAAPIClient:

    def get_service_name(self):
        return "Cloud Web App"

    def connect(self):
        print(f"Connecting to {self.get_service_name()}...")
        print("Connected!")
        return 0

    def get(self, url):
        print(f"Sending HTTP-GET request to {self.get_service_name()}: {url}")
        return 200

    def disconnect(self):
        print(f"Disconnecting from {self.get_service_name()}...")
        print("Disconnected!")
        return 0

File: JavaExample/src/main/java/hr/intis/cloud/examples/api/CWAAPIClient.java
package hr.intis.cloud.examples.api;

public class CWAAPIClient implements BasicAPIClient {

    public CWAAPIClient() {
    }

    @Override
    public String getServiceName() {
        return "Cloud Web App";
    }

    @Override
    public int connect() {
        System.out.printf("Connecting to %s...\n", getServiceName());
        System.out.println("Connected!");
        return 0;
    }

    @Override
    public int get(String url) {
        System.out.printf("Sending HTTP-GET request to %s: %s\n", getServiceName(), url);
        return 200;
    }

    @Override
    public int disconnect() {
        System.out.printf("Disconnecting from %s...\n", getServiceName());
        System.out.println("Disconnected!");
        return 0;
    }
}
